File: docs/vsto/codesnippet/CSharp/trin_wordcontentcontrolreference/RichText.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Microsoft.VisualStudio.Tools.Applications.Runtime;
using Word = Microsoft.Office.Interop.Word;
using Office = Microsoft.Office.Core;

namespace Trin_ContentControlReference
{
    public partial class ThisDocument
    {

        // RichTextContentControl constructor #1: create a RichTextContentControl at the current selection.
        //<Snippet700>
        private Microsoft.Office.Tools.Word.RichTextContentControl richTextControl1;

        private void AddRichTextControlAtSelection()
        {
            this.Paragraphs[1].Range.InsertParagraphBefore();
            this.Paragraphs[1].Range.Select();

            richTextControl1 = this.Controls.AddRichTextContentControl("richTextControl1");
            richTextControl1.PlaceholderText = "Enter your first name";
        }
        //</Snippet700>

        // RichTextContentControl constructor #2: create a RichTextContentControl at a specified range.
        //<Snippet701>
        private Microsoft.Office.Tools.Word.RichTextContentControl richTextControl2;

        private void AddRichTextControlAtRange()
        {
            this.Paragraphs[1].Range.InsertParagraphBefore();

            richTextControl2 = this.Controls.AddRichTextContentControl(this.Paragraphs[1].Range,
                "richTextControl2");
            richTextControl2.PlaceholderText = "Enter your first name";
        }
        //</Snippet701>

        // RichTextContentControl constructor #3: create a RichTextContentControl for every
        // native rich text control in the document.
        //<Snippet702>
        private System.Collections.Generic.List
            <Microsoft.Office.Tools.Word.RichTextContentControl> richTextControls;

        private void CreateRichTextControlsFromNativeControls()
        {
            if (this.ContentControls.Count <= 0)
                return;

            richTextControls = new System.Collections.Generic.List
                <Microsoft.Office.Tools.Word.RichTextContentControl>();
            int count = 0;

            foreach (Word.ContentControl nativeControl in this.ContentControls)
            {
                if (nativeControl.Type ==
                    Microsoft.Office.Interop.Word.WdContentControlType.wdContentControlRichText)
                {
                    count++;
                    Microsoft.Office.Tools.Word.RichTextContentControl tempControl =
                        this.Controls.AddRichTextContentControl(nativeControl,
                        "VSTORichTextControl" + count.ToString());
                    richTextControls.Add(tempControl);
                }
            }
        }
        //</Snippet702>


        // RichTextContentControl constructor #3: create a RichTextContentControl for every
        // native rich text control added to the document.
        //<Snippet703>
        void ThisDocument_RichTextContentControlAfterAdd(Word.ContentControl NewContentControl, bool InUndoRedo)
        {
            if (NewContentControl.Type == Word.WdContentControlType.wdContentControlRichText)
            {
                this.Controls.AddRichTextContentControl(NewContentControl,
                    "RichTextControl" + NewContentControl.ID);
            }
        }
        //</Snippet703>
    }
}

File: docs/vsto/codesnippet/VisualBasic/trin_contentcontrolreference/RichText.vb
Partial Public NotInheritable Class ThisDocument

    ' RichTextContentControl constructor #1: create a RichTextContentControl at the current selection.
    '<Snippet700>
    Dim richTextControl1 As Microsoft.Office.Tools.Word.RichTextContentControl

    Private Sub AddRichTextControlAtSelection()
        Me.Paragraphs(1).Range.InsertParagraphBefore()
        Me.Paragraphs(1).Range.Select()
        richTextControl1 = Me.Controls.AddRichTextContentControl("richTextControl1")
        richTextControl1.PlaceholderText = "Enter your first name"
    End Sub
    '</Snippet700>

    ' RichTextContentControl constructor #2: create a RichTextContentControl at a specified range.
    '<Snippet701>
    Dim richTextControl2 As Microsoft.Office.Tools.Word.RichTextContentControl

    Private Sub AddRichTextControlAtRange()
        Me.Paragraphs(1).Range.InsertParagraphBefore()
        richTextControl2 = Me.Controls.AddRichTextContentControl(Me.Paragraphs(1).Range, _
            "richTextControl2")
        richTextControl2.PlaceholderText = "Enter your first name"
    End Sub
    '</Snippet701>

    ' RichTextContentControl constructor #3: create a RichTextContentControl for every
    ' native rich text control in the document.
    '<Snippet702>
    Private richTextControls As New System.Collections.Generic.List _
            (Of Microsoft.Office.Tools.Word.RichTextContentControl)

    Private Sub CreateRichTextControlsFromNativeControls()
        If Me.ContentControls.Count <= 0 Then
            Return
        End If

        Dim count As Integer = 0
        For Each nativeControl As Word.ContentControl In Me.ContentControls
            If nativeControl.Type = Word.WdContentControlType.wdContentControlRichText Then
                count += 1
                Dim tempControl As Microsoft.Office.Tools.Word.RichTextContentControl = _
                    Me.Controls.AddRichTextContentControl(nativeControl, _
                    "VSTORichTextContentControl" + count.ToString())
                richTextControls.Add(tempControl)
            End If
        Next nativeControl
    End Sub
    '</Snippet702>

    ' RichTextContentControl constructor #3: create a 
    ' RichTextContentControl for every native rich text control added to the document.
    '<Snippet703>
    Private Sub ThisDocument_RichTextContentControlAfterAdd(ByVal NewContentControl As Word.ContentControl, _
        ByVal InUndoRedo As Boolean) Handles Me.ContentControlAfterAdd

        If NewContentControl.Type = Word.WdContentControlType.wdContentControlRichText Then
            Me.Controls.AddRichTextContentControl(NewContentControl, _
                "RichTextControl" + NewContentControl.ID)
        End If
    End Sub
    '</Snippet703>

End Class
